src/mongocxx/test/collection.cpp

```diff
@@ -35,14 +35,14 @@ TEST_CASE("collection copy", "[collection]") {
 
     std::string collname{"foo"};
     std::string collname2{"bar"};
-    collection coll = db[collname];
+    collection coll = db[stdx::string_view{collname}];
```

Comment:
Agreed, this isn't ideal.  Right now string::view_or_value has the same semantics as document::view_or_value, you can either move in a value-y type, or use a view.

If we want to allow users to just pass in a std::string without moving it, I think we'd want to copy the string and own the copy in the view_or_value.  Does that sound reasonable?  The other option, converting to a string_view, would lead to the same ownership issues that got us here in the first place.